File: array/best-time-to-buy-and-sell-stock.js
/**
 * https://leetcode.com/problems/best-time-to-buy-and-sell-stock
 * 
 * @param {number[]} prices
 * @return {number}
 */

const maxProfit = (prices) => {
    let lowestDay = Infinity;
    let highestDay = 0;
    
    for (i = 0; i < prices.length; i++) {
        lowestDay = Math.min(lowestDay, prices[i]);
        highestDay = Math.max(highestDay, prices[i] - lowestDay);
    }
    
    return highestDay;

}; 


//Brute force with two for loops, for every value in prices calculate value - value[i] to the right of it

File: python/sliding-window/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        lowest = prices[0]
        max_prof = 0

        for price in prices:
            lowest = min(lowest, price)
            
            max_prof = max(max_prof, price - lowest)

        return max_prof



# https://leetcode.com/problems/best-time-to-buy-and-sell-stock/
# Actually found pretty tricky as haven't done in a while, question
# is actually two pointers with the first pointer always being the lowest 
# value and being checked on each iteration.
            

            

        